src/components/Common/TaxInputs/TaxInputs.tsx

```diff
@@ -13,15 +13,47 @@ const dompurifyConfig = { ALLOWED_TAGS: ['a', 'b', 'strong'], ALLOWED_ATTR: ['ta
 interface EmpQ {
   question: NonNullable<EmployeeStateTaxQuestion>
   requirement?: never
+  isDisabled?: boolean
 }
 interface CompR {
   requirement: TaxRequirement
   question?: never
+  isDisabled?: boolean
 }
 
-type NumberFieldProps = { isCurrency?: boolean }
+type NumberFieldProps = { isCurrency?: boolean; isPercent?: boolean }
+
+export function QuestionInput({
+  questionType,
+  ...props
+}: (EmpQ | CompR) & {
+  questionType: string
+}) {
+  switch (questionType.toLowerCase()) {
+    case 'date':
+      return <DateField {...props} />
+    case 'radio':
+      return <RadioInput {...props} />
+    case 'text':
+    case 'account_number': //TODO: temporary - need special handling for account numbers
+      return <TextInput {...props} />
+    case 'select':
+      return <SelectInput {...props} />
+    case 'number':
+      return <NumberInput {...props} />
+    case 'workers_compensation_rate':
+    case 'percent':
+    case 'tax_rate':
+      return <NumberInput {...props} isPercent />
+    case 'currency':
+      return <NumberInput {...props} isCurrency />
+    default:
+      return <TextInput {...props} />
+    // return null
+  }
+}
 
-export function SelectInput({ question, requirement }: EmpQ | CompR) {
```

Comment:
Should inputs have their own file or files instead of all living in the TaxInputs module?